multinode-demo/setup.sh

```diff
@@ -1,11 +1,54 @@
 #!/bin/bash
 
-num_tokens=${1:-1000000000}
+num_tokens=1000000000
+public_ip=
 
 here=$(dirname "$0")
 # shellcheck source=multinode-demo/common.sh
 source "$here"/common.sh
 
+usage () {
+  cat <<EOF
+usage: $0 [-n num_tokens] [-P] [-p public_ip_address]
+
+Creates a fullnode configuration
+
+ -n num_tokens         - Number of tokens to create
+ -p public_ip_address  - Public IP address to advertise
+                         (default uses the system IP address, which may be
+                          on a private network)
+ -P                    - Autodetect the public IP address of the machine
+EOF
+}
+
+while getopts "h?n:p:P" opt; do
```

Comment:
there are some error cases you might handle here, but it's not a gating issue, IMHO

Reply:
Oh like what?  I’ll fix

Reply:
I always have to go look at my old getopts() calls along with the bash manpage...

Reply:
https://github.com/cornholington/autocross/blob/master/results/esc.sh captures my latest thinking on getopts().  note the initial ':' in the string and the unknown opt case statement